### DISCUSSION TITLE
How to create record via url.

### DISCUSSION THREAD

**Discussion by kenyonbowers** [pocketbase/pocketbase#1618] [Q&A] [answered]
So I am trying to make a PocketBase app thing, and I want to use Vanilla JS for the frontend (please don't ask why). So I am able to get records from my PocketBase and I am able to list them, but I can't get it to create records.
```
fetch('https://mypocketbase.pockethost.io/api/collections/testing/records', {
    method: 'POST',
    headers: {
        'Accept': "application/json",
        'Content-Type': 'multipart/form-data',
    },
    body: {
        "name": "Hello World!"
    }
})
```
I checked the admin logs and found this message:
```
{
  "errorDetails": "no multipart boundary param in Content-Type",
  "errorMessage": "Failed to load the submitted data due to invalid formatting."
}
````
I don't know what to do. I've tried looking up what the "boundary" thing is but I don't know and I am all out of ideas.

**Comment by kenyonbowers** [comment, marked as answer]
I got it to work with this:
```
 fetch('https://mypocketbase.pockethost.io/api/collections/testing/records', {
            method: "POST",
            body: JSON.stringify({ name: "Hello World!" }),
            headers: {
                "Content-Type": "application/json"
            }
        })
```

**Reply by ganigeorgiev** [reply]
_Just for info, you can use the [JS SDK](https://github.com/pocketbase/js-sdk) with plain js just fine, it doesn't have a dependency on any framework and internally uses the native `fetch` for sending requests. For loading the SDK with a regular script src you can manually download the generated /dist UMD bundle or use a CDN: https://cdn.jsdelivr.net/gh/pocketbase/js-sdk@master/dist/pocketbase.umd.js. Modern browsers have also [support for modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules) so you can also use the ESM bundle._

**Reply by kenyonbowers** [reply]
I saw that file, however, I have no idea how to use it because when loading it in the src of a script element, where would I even go from there? Like how do I use it? Do I need another script element? am I able to access PocketBase inside it?

**Reply by ganigeorgiev** [reply]
> Do I need another script element?

Yes, the UMD bundle will register `PocketBase` in the global context so you should be able to access it from anywhere after initialization, eg:

```html
<script src="https://cdn.jsdelivr.net/gh/pocketbase/js-sdk@master/dist/pocketbase.umd.js"></script>
<script>
    const pb = new PocketBase("http://127.0.0.1:8090");
    
    pb.collection("example").then((result) => {
        console.log(result);
    }).catch((err) => {
        console.log(err);
    })
</script>
```
OR if you are using ES modules:

```html
<script type="module">
    // you can download the file from https://github.com/pocketbase/js-sdk/blob/master/dist/pocketbase.es.mjs
    import PocketBase from "./path/to/dist/pocketbase.es.mjs";

    const pb = new PocketBase("http://127.0.0.1:8090");
    
    pb.collection("example").then((result) => {
        console.log(result);
    }).catch((err) => {
        console.log(err);
    })
</script>
```

**Reply by kenyonbowers** [reply]
Wow! That's brilliant! Thank you so much for the information.
